packages/rpc/src/util.ts

```diff
@@ -1,40 +1,69 @@
-import { mbjs, RPC_ENDPOINTS, NEAR_RPC_ENDPOINTS } from '@mintbase-js/sdk';
 import fetch from 'cross-fetch';
 
+export type RPC_OPTIONS  = 'lava' | 'near' | 'beta' | 'fastnear' | 'pagoda'
 
-export type RPC_OPTIONS  = 'lava' | 'near' | 'beta'
+type RpcNodes = {
+  [key in RPC_OPTIONS]?: string;
+};
 
+export const rpcNodes: {
+  mainnet: RpcNodes;
+  testnet: RpcNodes;
+} = {
+  mainnet: {
+    lava: "https://g.w.lavanet.xyz:443/gateway/near/rpc-http/f538cb3b0a85aafdb9996886d004ee0a",
+    near: "https://rpc.mainnet.near.org",
+    fastnear: "https://free.rpc.fastnear.com/",
+    pagoda: "https://rpc.mainnet.pagoda.co"
+  },
+  testnet: {
+    lava: "https://g.w.lavanet.xyz:443/gateway/neart/rpc-http/f538cb3b0a85aafdb9996886d004ee0a",
+    near: "https://rpc.testnet.near.org",
+    pagoda: "https://rpc.testnet.pagoda.co",
+    fastnear:
+      "https://test.rpc.fastnear.com",
+  },
+};
 
 export const requestFromNearRpc = async (
   body: Record<string, unknown>,
-  network?: string,
-  rpc?:  RPC_OPTIONS,
-): Promise<{ result: Record<string, unknown>, error: unknown } | undefined> => {
+  rpcUrl?: string): Promise<{ result: Record<string, unknown>, error: unknown } | undefined> => {
 
-  const fetchUrl =  mbjs.keys.nearRpcUrl || RPC_ENDPOINTS[mbjs.keys.rpc][mbjs.keys.network]  || NEAR_RPC_ENDPOINTS[mbjs.keys.network];
-  const rpcAddress = network && rpc ? RPC_ENDPOINTS[rpc][network] : fetchUrl;
+   const validUrls = Object.values(rpcNodes.mainnet).concat(Object.values(rpcNodes.testnet));
 
-  const res = await fetch(rpcAddress, {
-    method: 'POST',
-    body: JSON.stringify(body),
-    headers: { 'Content-type': 'application/json' },
-  });
+  // Set default rpcUrl if not provided
+   if (!rpcUrl) {
+    rpcUrl = rpcNodes.mainnet.near;
+  }
+
+  if (rpcUrl && !validUrls.includes(rpcUrl)) {
+    throw new Error('Invalid rpcUrl');
+  }
```

Comment:
Why should we curate a list here, what if someone wants to use their own private RPC?